pkgs/by-name/bl/bloodhound-ce/package.nix

```diff
@@ -0,0 +1,227 @@
+{
+  buildGoModule,
+  fetchFromGitHub,
+  fetchurl,
+  lib,
+  makeWrapper,
+  nodejs_22,
+  p7zip,
+  python311,
+  runCommand,
+  stdenv,
+  yarn-berry_3,
+}:
+let
+  inherit (lib) concatLines flip pipe;
+  inherit (lib.versions) major minor patch;
+
+  # reference: https://github.com/SpecterOps/BloodHound/blob/v8.3.1/dockerfiles/bloodhound.Dockerfile
+
+  pname = "bloodhound-ce";
+  version = "8.3.1";
+
+  src = fetchFromGitHub {
+    owner = "SpecterOps";
+    repo = "BloodHound";
+    tag = "v${version}";
+    hash = "sha256-mIoQkxUxv2BktFGSLKo5RVEF/7JByiyCWC2o9GWS9w4=";
+  };
+
+  filter = flip pipe [
+    (map (x: ''
+      mkdir -p "$out/${dirOf x}"
+      cp -r "${src}/${x}" "$out/${x}"
+    ''))
+    concatLines
+    (runCommand "filtered-source" { })
+  ];
+
+  yarn-berry = yarn-berry_3;
+
+  ##############################################################################
+
+  frontend = stdenv.mkDerivation (finalAttrs: {
+    pname = "bloodhound-ce-frontend";
+    inherit version;
+
+    src = filter [
+      ".yarn/plugins"
+      ".yarn/releases"
+      ".yarnrc.yml"
+      "cmd/ui"
+      "constraints.pro"
+      "package.json"
+      "packages/javascript"
+      "yarn-workspaces.json"
+      "yarn.lock"
+    ];
```

Comment:
I would not recommend to filter files like this as this could easily miss some configuration files and in the end we rebuild all packages with every update anyway.